src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureDtdProcessingInApiDesign.cs

```diff
@@ -521,59 +474,48 @@ private void AnalyzeSymbolForXmlTextReaderDerivedType(SymbolAnalysisContext cont
                 // it is not setting the actual property for this class.
 
                 // The goal is to find all assignment like in the example above, "this.xxx.xxx.Property = ...;".
-                // For simplicity, here we adopt a simple but inaccurate logic:
-                //   If the target is a member access node, then the only pattern we are looking for is "this.Property"
-                SyntaxNode memberAccessNode = _syntaxNodeHelper.GetDescendantMemberAccessExpressionNodes(left).FirstOrDefault();
-
-                // if assignment target doesn't have any member access node,
-                // then we treat it as an instance property access without explicit 'this' ('Me' in VB)
-                if (memberAccessNode == null)
+                if (propertyReference.Instance == null
+                    || propertyReference.Instance.IsImplicit)
                 {
                     //stop here, to avoid false positive, as long as there's one setting <Property> to secure value, we are happy
                     return isIntendedValue;
                 }
 
-                SyntaxNode exp = _syntaxNodeHelper.GetMemberAccessExpressionNode(memberAccessNode);
-                ISymbol expSymbol = SyntaxNodeHelper.GetSymbol(exp, model);
-
-                isTargetProperty = expSymbol.Kind == SymbolKind.Parameter && ((IParameterSymbol)expSymbol).IsThis;
-                if (!isTargetProperty)
-                {
-                    return false;
-                }
-
-                SyntaxNode name = _syntaxNodeHelper.GetMemberAccessNameNode(memberAccessNode);
-                ISymbol nameSymbol = SyntaxNodeHelper.GetSymbol(name, model);
-
-                isTargetProperty = isTargetPropertyFunc(nameSymbol as IPropertySymbol);
-                if (!isTargetProperty)
+                if (propertyReference.Instance.Kind != OperationKind.InstanceReference)
                 {
                     return false;
                 }
 
                 // stop here, same reason as stated above
-                return isIntendedValue;
+                return isTargetProperty && isIntendedValue;
             }
+        }
 
-            public static Diagnostic CreateDiagnostic(
-                Location location,
-                DiagnosticDescriptor rule,
-                params object[] args)
+        private class ThreadSafeBoolean
```

Comment:
I don't like to work with `Interlocked.CompareExchange` so I have created a little helper. It seems that on `master` I did the same and called it `AtomicBoolean`, let me know the preferred name.